backend/inspirehep/records/api/mixins.py

```diff
@@ -587,3 +590,79 @@ def get_modified_experiment_uuids(self):
         pids_changed = set.symmetric_difference(set(pids_latest), set(pids_previous))
 
         return list(self.get_records_ids_by_pids(list(pids_changed)))
+
+
+class StudentsAdvisorMixin:
+    def generate_entries_for_table(self):
+        table_entries_buffer = []
+        student_record_uuid = self.id
+        for advisor in self.get_value("advisors", []):
+            if "record" not in advisor:
+                LOGGER.info(
+                    f"Skipping creating entries in "
+                    f"{StudentsAdvisors.__tablename__} table. Advisor record is missing",
+                    recid=self.get("control_number"),
+                    uuid=str(self.id),
+                )
+                continue
+            advisor_recid = get_recid_from_ref(advisor["record"])
+            advisor_record_uuid = (
+                PersistentIdentifier.query.with_entities(
+                    PersistentIdentifier.object_uuid
+                )
+                .filter_by(pid_value=str(advisor_recid), pid_provider="recid")
+                .scalar()
```

Comment:
you're doing one query per advisor here. It would be better to do one query per record instead. Why do you need to write this query BTW, don't we have helpers already to get a list of record UUIDs given a list of PIDs?

Reply:
Does it matter so much? How many advisors we might have per record? Of course I can make one query, but then I'll have to do some transformations to retrieve refs and then match query results with advisors cause an advisor might not have `$ref`.

No, we have only `get_records_by_pids` which returns all the metadata.

Reply:
OK, I see how it's a little bit more work. I don't know if it's worth it. I think so, but I'll let @drjova decide. If not, you should probably use `pid_type` instead of `pid_provider` (everywhere).